Enterprise-CMCS/macpro-onemac
# Oy2 23917 - add new forms and config for submitting 1915b waivers

Story: https://qmacbis.atlassian.net/browse/OY2-23917
Endpoint: See github-actions bot comment

### Details

Add different forms and config for each of the 1915b waiver types so that they are seperated by waiverAuthority. Each waiver type and waiver authority type combination has a different set of required and optional attachments.

### Changes

- Split each of the waiver forms (Iniitial, Renewal, Amendment) into 2 forms, one for (b)(4) and one for b other.
- Adapted submitXXXXX endpoints for the 3 waiver types to determine configuration for validation and submittal based on the submitted waiver authority

### Implementation Notes

- I didnt go as far as creating new "types" since that would require far more extensive changes as well as some UI impacts that were not part of the story. This is why I altered the api submission classes to determine the proper config rather than creating new submission endpoints as the submission endpoints work largely off of submission type

### Test Plan

1. Login as state submitter
2. Submit one each of the 6 new forms - for each one validate required attachments match the ACs
3. Progress a new waiver submission to RAI Issued status and verify you can still submit an RAI response
4. Progress a new waiver submission to In Review and verify you can still withdraw
